packages/core/src/interpreter.ts

```diff
@@ -1183,7 +1183,7 @@ export class Interpreter<
   }
   public spawnMachine<
     TChildContext,
-    TChildStateSchema,
+    TChildStateSchema extends StateSchema,
```

Comment:
This was a bug in the code. Since TS became more conservative now, when it comes to such unconstrained generics, it started to complain about this.